addons/website_event_booth_sale/models/sale_order.py

```diff
@@ -23,33 +23,47 @@ def _cart_find_product_line(self, product_id=None, line_id=None, **kwargs):
         return lines
 
     def _website_product_id_change(self, order_id, product_id, qty=0, **kwargs):
+        if self._context.get('pricelist') != self.pricelist_id.id:
+            self = self.with_context(pricelist=self.pricelist_id.id)
+
         values = super(SaleOrder, self)._website_product_id_change(order_id, product_id, qty=qty, **kwargs)
         event_booth_pending_ids = kwargs.get('event_booth_pending_ids')
         if event_booth_pending_ids:
             order_line = self.env['sale.order.line'].sudo().search([
                 ('id', 'in', self.order_line.ids),
                 ('event_booth_pending_ids', 'in', event_booth_pending_ids)])
             booths = self.env['event.booth'].browse(event_booth_pending_ids)
-            new_registration_ids = [Command.create({
-                                        'event_booth_id': booth.id,
-                                        **kwargs.get('registration_values'),
-                                    }) for booth in booths]
-            if order_line:
-                event_booth_registration_ids = [Command.delete(reg.id)
-                                                for reg in order_line.event_booth_registration_ids] + new_registration_ids
-            else:
-                event_booth_registration_ids = new_registration_ids
+            if order_line.event_booth_pending_ids.ids != event_booth_pending_ids:
+                new_registration_ids = [Command.create({
+                                            'event_booth_id': booth.id,
+                                            **kwargs.get('registration_values'),
+                                        }) for booth in booths]
+                if order_line:
+                    event_booth_registration_ids = [Command.delete(reg.id)
+                                                    for reg in order_line.event_booth_registration_ids] + new_registration_ids
+                else:
+                    event_booth_registration_ids = new_registration_ids
+                values.update(event_booth_registration_ids=event_booth_registration_ids)
 
+            if self.pricelist_id.discount_policy == 'without_discount':
+                price_unit = sum(booth.price for booth in booths)
+            else:
+                price_unit = sum(booth.price_reduce for booth in booths)
             values.update(
                 event_id=booths.event_id.id,
-                event_booth_registration_ids=event_booth_registration_ids,
+                price_unit=price_unit,
                 name=booths._get_booth_multiline_description,
             )
 
         return values
 
     def _cart_update(self, product_id=None, line_id=None, add_qty=0, set_qty=0, **kwargs):
         values = {}
+        if line_id:
+            line = self.env['sale.order.line'].browse(line_id)
+            booths = line.event_booth_pending_ids
+            if booths:
+                kwargs.update({'event_booth_pending_ids': booths.ids})
```

Comment:
Updating the kwargs manually before calling super is a bit strange.
Also feels strange that we need to store that information into kwargs as it's already present on the so.line that is passed further down the call chain anyway.

Reply:
You mean in the "update_pricelist" use case?
I see that in that case it also passes the product and "add_qty=0".

So maybe instead we could add it there:
```py
if product.detailed_type == 'event_booth' and set_qty > 1:
    set_qty = 1
    values['warning'] = _('You cannot manually change the quantity of an Event Booth product.')
elif product.detailed_type == 'event_booth' and line_id and add_qty == 0 and not kwargs.get('event_booth_pending_ids'):
    # when updating the pricelist, the website_sale module call this "_cart_update" method without passing the event.booth we need to update in the kwargs
    # -> we manually set the argument to make sure the price is updated in "_website_product_id_change"
    kwargs['event_booth_pending_ids'] = self.env['sale.order.line'].browse(line_id).event_booth_pending_ids.ids
```

Maybe that is not correct though, you would need to double-check (and a unit test would be ideal of course ^^).